presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/NoSubqueryExpressionLeftSanityChecker.java

```diff
@@ -0,0 +1,77 @@
+package com.facebook.presto.sql.planner.sanity;
+
+import com.facebook.presto.sql.analyzer.SemanticErrorCode;
+import com.facebook.presto.sql.analyzer.SemanticException;
+import com.facebook.presto.sql.planner.SimplePlanVisitor;
+import com.facebook.presto.sql.planner.plan.FilterNode;
+import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.facebook.presto.sql.planner.plan.ProjectNode;
+import com.facebook.presto.sql.planner.plan.TableScanNode;
+import com.facebook.presto.sql.planner.plan.ValuesNode;
+import com.facebook.presto.sql.tree.DefaultTraversalVisitor;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.SubqueryExpression;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+
+public class NoSubqueryExpressionLeftSanityChecker
+        implements PlanSanityChecker
+{
+    @Override
+    public void validate(PlanNode plan)
+    {
+        for (Expression expression : ExpressionExtractor.extractExpressions(plan)) {
+            new DefaultTraversalVisitor<Void, Void>()
+            {
+                @Override
+                protected Void visitSubqueryExpression(SubqueryExpression node, Void context)
+                {
+                    throw new SemanticException(SemanticErrorCode.NOT_SUPPORTED, node, "Given subquery is not yet supported");
```

Comment:
Plan sanity checkers are about enforcing invariants. I.e., make sure optimizers are not breaking assumptions and producing incorrect plans. So, they should throw IllegalArgumentExceptions instead of SemanticExceptions (the analyzer should be throwing these)